src/scheduler/gc_work.rs

```diff
@@ -108,6 +108,8 @@ impl<C: GCWorkContext> Release<C> {
 impl<C: GCWorkContext + 'static> GCWork<C::VM> for Release<C> {
     fn do_work(&mut self, worker: &mut GCWorker<C::VM>, mmtk: &'static MMTK<C::VM>) {
         trace!("Release Global");
+        self.plan.base().gc_trigger.policy.on_gc_release(mmtk);
```

Comment:
Not a problem for now.  This is added before the `VMCollection::vm_release()` below, but `plan_mut.release` happens after `VMCollection::vm_release()`.  It's a bit weird this way, but it doesn't matter if the VM doesn't implement `vm_release`.

 `VMCollection::vm_release()` was created for VMs to do what our current `RefEnqueue` does to handle weak references.  I am considering moving `VMCollection::vm_release()` to a dedicated work packets, because ref-enqueuing work does not depend on the `Plan` instance or the release of the `Plan`, and can therefore be parallelised.

Reply:
If we can run some code when we open the `Release` bucket, that would be a good timing to run this `on_gc_release()` method.